### PR TITLE
Allow intercepted request to be used by requestIdFactory function.

### PR DESCRIPTION
**Pull request by marciopd**
Refs: #37 #42 

Hello,

This PR allows `requestIdFactory` to receive the intercepted request as an input for generating the id.

Tests are provided for each different server supported and the docs were updated too.

Please let me know if something needs to be changed or improved.

Best regards!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -295,8 +295,10 @@ These are the available config options for the middleware/plugin functions. All
   // Used if useHeader/echoHeader is set to true.
   headerName: 'X-Request-Id',
   // A custom function to generate your request ids (default: UUID v1).
+  // The function will receive the intercepted request (as-is from the server being used) as its first argument.
```
Comment (puzpuzpuz) [inline README.md:298]:
```suggestion
  // The function will receive the intercepted request (as-is from the framework being used) as its first argument.
```

File: README.md [hunk 2]
```diff
@@ -295,8 +295,10 @@ These are the available config options for the middleware/plugin functions. All
...
+  // The returned id could be an usual string or any custom object, like in the example below.
```
Comment (puzpuzpuz) [inline README.md:299]:
```suggestion
  // The returned id could be a usual string, or number, or any custom object, like in the example below.
```

File: README.md [hunk 3]
```diff
@@ -295,8 +295,10 @@ These are the available config options for the middleware/plugin functions. All
...
   // Ignored if useHeader is set to true.
-  requestIdFactory: () => 'Your request id',
+  requestIdFactory: (request) => { return { id: 'Your request id', client_ip: request.headers.client_ip } },
```
Comment (puzpuzpuz) [inline README.md:301]:
```suggestion
  requestIdFactory: (req) => ({
    id: 'Your request id',
    customHeader: req.headers['X-Custom-Header']
  }),
```

**Review by puzpuzpuz — COMMENTED** [review]
LGTM in general. Could you also include an update for TS typings?
https://github.com/puzpuzpuz/cls-rtracer/blob/master/index.d.ts#L3

**Comment by marciopd** [comment]
Hello @puzpuzpuz , just updated the code with the suggestions. 

Please let me know if something else needs attention. Best regards!

**Review by puzpuzpuz — APPROVED** [review]
LGTM. Many thanks for your contribution!

**Pull request merged**